### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mong2**
https://github.com/cloudpassage/cortex/issues/22

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -45,7 +45,6 @@ RUN pip install \
 
 # Drop in the app code
 COPY app/ /app/
-COPY cortex_conf.yml /
```
Comment (eatwithforks) [inline Dockerfile]: Is the intent not to have don-bot run by itself?
docker build . will not have local.env variables set. So the image won't have any the cortex_conf variables

File: app/cortexlib/ipblockcheck.py [hunk 2]
```diff
@@ -60,3 +53,13 @@ def validate_config(self):
             if not isinstance(v, str):
                 msg = "%s is not the correct type!  Should be string!" % str(v)
                 raise ValueError(msg)
+
```
Comment (eatwithforks) [inline app/cortexlib/ipblockcheck.py]: These two functions, string_list and string_bool are also in the other test. Should make into a lib so there's no duplicate code

File: app/cortexlib/quarantine.py [hunk 3]
```diff
@@ -50,3 +44,13 @@ def validate_config(self):
             if not isinstance(v, str):
                 msg = "%s is not the correct type!  Should be string!" % str(v)
                 raise ValueError(msg)
+
```
Comment (eatwithforks) [inline app/cortexlib/quarantine.py]: duplicate code, consider make into a lib

File: app/cortexlib/ipblockcheck.py [hunk 4]
```diff
@@ -36,13 +32,10 @@ def extract_ip_from_event(cls, event):
...
         return None
 
-
     def set_ipblockcheck_config(self):
-        with open(IpBlockCheck.config_file, 'r') as config:
-            ipblock_conf = yaml.load(config)["ipblocker"]
-        self.ip_zone_name = ipblock_conf["ip_zone_name"]
-        self.trigger_events = ipblock_conf["trigger_events"]
-        self.trigger_only_on_critical = ipblock_conf["trigger_only_on_critical"]
+        self.ip_zone_name = os.getenv("IPBLOCKER_IP_ZONE_NAME")
```
Comment (eatwithforks) [inline app/cortexlib/ipblockcheck.py]: set_config is similar in the other test. Is there a way to do it via a lib that'll set the variables in the init.

File: app/cortexlib/quarantine.py [hunk 5]
```diff
@@ -26,12 +22,10 @@ def should_quarantine(self, event):
         return False
 
     def set_quarantine_config(self):
-        with open(Quarantine.config_file, 'r') as config:
-            quar_conf = yaml.load(config)["quarantine"]
-        self.trigger_group_names = quar_conf["trigger_group_names"]
-        self.quarantine_group_name = quar_conf["quarantine_group_name"]
-        self.trigger_events = quar_conf["trigger_events"]
-        self.trigger_only_on_critical = quar_conf["trigger_only_on_critical"]
+        self.trigger_group_names = self.string_list(os.getenv("QUARANTINE_TRIGGER_GROUP_NAME"))
```
Comment (eatwithforks) [inline app/cortexlib/quarantine.py]: consider into lib

File: app/test/integration/test_integration_cortexlib_quarantine.py [hunk 6]
```diff
@@ -43,10 +46,13 @@ def test_trigger_validation_fail(self):
...
     def test_quarantine_event_trigger(self):
-        q_grp = "Quarantine"
         q = self.instantiate_cortexlib_quarantine()
+        quar_event["server_group_name"] = q.trigger_group_names[0]
+        quar_event["type"] = q.trigger_events[0]
+        q_grp = q.quarantine_group_name
         assert q.should_quarantine(quar_event)["quarantine_group"] == q_grp
 
     def test_quarantine_event_no_trigger(self):
+        print os.getenv("IPBLOCKER_ENABLED")
```
Comment (eatwithforks) [inline app/test/integration/test_integration_cortexlib_quarantine.py]: take out debug print

File: Dockerfile (hunk 1 above)
Comment (mong2) [reply Dockerfile]: local.env is in test/configuration/local.env. Test will load the file

Comment (eatwithforks) [reply Dockerfile]: ok. then should add to the readme what env vars to set if user wants to run it as a standalone. and/or point to the local.env file with instructions.

Comment (mong2) [reply Dockerfile]: i thought the whole point is not to let customer run it standalone, no? that's why i have local.env only for pass docker build.

Comment (eatwithforks) [reply Dockerfile]: ok i get that but its confusing to have all those instructions in the readme if this isn't meant to be run as a standalone. At the very least, remove those instructions, point to cortex for instructions, and specify which tag customer should fall back to if they want to use it as standalone

File: app/cortexlib/quarantine.py [hunk 7]
```diff
@@ -1,52 +1,22 @@
...
-        self.trigger_only_on_critical = True
-        self.set_quarantine_config()
-        self.validate_config()
+        self.config = ConfigHelper()
 
     def should_quarantine(self, event):
         """Returns an enriched event object, or False if the event is OK"""
-        event["quarantine_group"] = self.quarantine_group_name
-        if (self.trigger_only_on_critical == True and
+        print self.config
```
Comment (eatwithforks) [inline app/cortexlib/quarantine.py]: leftover debug print should be removed

**Comment by eatwithforks** [comment]
+1. leave readme to other issue

**Pull request merged**